src/main/java/io/supertokens/session/Session.java

```diff
@@ -166,6 +170,16 @@ public static SessionInformationHolder createNewSession(TenantIdentifier tenantI
             if (userIdMappings.containsKey(recipeUserId)) {
                 recipeUserId = userIdMappings.get(recipeUserId);
             }
+
+            if(semVer!= null && semVer.greaterThanOrEqualTo(SemVer.v5_2)) {
```

Comment:
simply use a boolean whether to do this check or not

Reply:
done